docs/ADDRESS_LOOKUP.md

```diff
@@ -11,7 +11,63 @@
 - If you're integrating with Drop-in:
     - Implement the mandatory `onAddressLookupQueryChanged(query: String)` callback and optional `onAddressLookupCompletion(lookupAddress: LookupAddress)` callback.
     - Pass the result of these actions by using `AddressLookupDropInServiceResult` class.
+      ```kotlin
+      override fun onAddressLookupQueryChanged(query: String) {
+          // Use the query parameter to make a call to your desired maps endpoint 
+          // Map the address objects LookupAddress object and pass it back to Drop-in as follows:
+          val options: List<LookupAddress> = // mapped list of address objects
+          sendAddressLookupResult(AddressLookupDropInServiceResult.LookupResult(options))  
+      }
+
+      override fun onAddressLookupCompletion(lookupAddress: LookupAddress): Boolean {
+          // Optionally use this callback to get the complete details for a LookupAddress option selected by the shopper
+          // If you need to use this callback to make a call to get complete details for the selected address option you must return true.
+          // You can also use this function conditionally. In that case, return true if you perform an api call, false otherwise.
+          // Make a call to your desired maps endpoint and pass the complete LookupAddress object back to Drop-in as follows:
+          val shouldCompleteAddress = // whether an api call needs to be made or not
+          if (shouldCompleteAddress) {
+              val completeLookupAddress = // complete LookupAddress object
+              sendAddressLookupResult(AddressLookupDropInServiceResult.LookupComplete(completeLookupAddress))
+              return true
+          }
+          // If you do not need to use this callback either do not implement it, or return false.
+          return false
+      } 
+      ```
 - If you're integrating with standalone `CardComponent`:
     - Set `AddressLookupCallback` via `CardComponent.setAddressLookupCallback(AddressLookupCallback)` function to receive the related events.
     - Pass the result of these actions by calling `CardComponent.setAddressLookupResult(addressLookupResult: AddressLookupResult)`.
+    ```kotlin
+    cardComponent.setAddressLookupCallback(object : AddressLookupCallback {
+        override fun onQueryChanged(query: String) {
+            // Use the query parameter to make a call to your desired maps endpoint
+            // Map the address objects LookupAddress object and pass it back to Drop-in as follows:
+            val options: List<LookupAddress> = // mapped list of address objects
+            cardComponent.updateAddressLookupOptions(options)
+        }
+
+        override fun onLookupCompletion(lookupAddress: LookupAddress): Boolean {
```

Comment:
I feel like this code section and comment is too complex. I have a few suggestions, feel free to take any that you like:
- Add a small comment on top of the method, something like `// Optional: use this callback if you need to make an extra API call after the shopper selects an address option`. Makes it easy to decide whether you should continue reading the whole section or skip it.
- Write the whole sample assuming they will always make an API call if they implemented this method, and only add a comment next to `return true` that they can return false if they do not need to make the API call in certain cases. I find it easier when the docs focus on the most standard use case but mentions the extra options in case someone needs them.
- Make the comments a bit shorter 😅